packages/openapi/src/helpers.ts

```diff
@@ -164,3 +170,126 @@ export function isReadonlyProperty(program: Program, property: ModelProperty) {
   // readonly: true, but using separate schemas.
   return visibility?.length === 1 && visibility[0] === "read";
 }
+
+/**
+ * Determines if a OpenAPIExtensionKey is start with `x-`.
+ */
+export function isOpenAPIExtensionKey(key: string): key is ExtensionKey {
+  return key.startsWith("x-");
+}
+
+/**
+ * Validate that the given string is a valid URL.
+ * @param program  Program
+ * @param target Diagnostic target for any diagnostics that are reported
+ * @param url The URL to validate
+ * @param propertyName The name of the property that the URL is associated with
+ * @returns true if the URL is valid, false otherwise
+ */
+export function validateIsUri(
+  program: Program,
+  target: DiagnosticTarget,
+  url: string,
+  propertyName: string,
+): boolean {
+  try {
+    // Attempt to create a URL object from the given string. If
+    // successful, the URL is valid.
+    new URL(url);
+    return true;
+  } catch {
+    // If the URL is invalid, report a diagnostic with the given
+    // target, property name and value.
+    reportDiagnostic(program, {
+      code: "not-url",
+      target: target,
+      format: { property: propertyName, value: url },
+    });
+    return false;
+  }
+}
+
+/**
+ * Validate the AdditionalInfo model against a reference.
+ *
+ * This function checks that the properties of the given AdditionalInfo object
+ * are a subset of the properties defined in the AdditionalInfo model.
+ *
+ * @param program - The TypeSpec Program instance
+ * @param target - Diagnostic target for reporting any diagnostics
+ * @param typespecType - The AdditionalInfo object to validate
+ * @param reference - The reference string to resolve the model
+ * @returns true if the AdditionalInfo object is valid, false otherwise
+ */
+export function validateAdditionalInfoModel(
```

Comment:
A few issues
1. That name doesn't makes sense as it validate more than info model now
2. Just take the converted js object not the model(call `typespecValueToJson` to convert the model to json for the legacy case)